EIPS/eip_generic_token_upgrade_standard.md

```diff
@@ -0,0 +1,191 @@
+function computeDowngrade(uint256 destinationAmount) external view
+        returns (uint256 sourceAmount, uint256 destinationRemainder)
+```
+
+
+#### upgrade
+
+Upgrades the `amount` of source token to the destination token in the specified ratio. The destination tokens will be sent to the ```_to``` address. The function should lock the source tokens in the upgrade contract or burn them. If the downgrade Optional Ext. is implemented, the source tokens should be locked instead of burning. The function should `throw` if the caller's address does not have enough source token to upgrade or if ```isUpgradeActive``` is returning ```false```. The function MUST also fire the `Upgrade` event. `approve` Must be called first on the source contract. 
+``` js
+/// @dev A method to conduct an upgrade from source token to destination token.
+/// The call will fail if upgrade status is not true, if approve has not been called
+/// on the source contract, or if sourceAmount is larger than the amount of source tokens at the msg.sender address.
+/// If the ratio would cause an amount of tokens to be destroyed by rounding/truncation, the upgrade call will
+/// only upgrade the nearest whole amount of source tokens returning the excess to the msg.sender address. 
+/// Emits the Upgrade event
+/// @param _to The address the destination tokens will be sent to upon completion of the upgrade
+/// @param sourceAmount The amount of source tokens that will be upgraded 
+function upgrade(address _to, uint256 sourceAmount) external
+```
+
+
+#### downgrade (Optional Ext.)
+Downgrades the `amount` of destination token to the source token in the specified ratio. The source tokens will be sent to the ```_to``` address. The function should unwrap the destination tokens back to the source tokens. The function should `throw` if the caller's address does not have enough destination token to downgrade or if ```isDowngradeActive``` is returning ```false```. The function MUST also fire the `Downgrade` event. `approve` Must be called first on the destination contract. 
+``` js
+/// @dev A method to conduct a downgrade from destination token to source token.
+/// The call will fail if downgrade status is not true, if approve has not been called
+/// on the destination contract, or if destinationAmount is larger than the amount of destination tokens at the msg.sender address.
+/// If the ratio would cause an amount of tokens to be destroyed by rounding/truncation, the downgrade call will only downgrade
+/// the nearest whole amount of destination tokens returning the excess to the msg.sender address. 
+///  Emits the Downgrade event
+/// @param _to The address the source tokens will be sent to upon completion of the downgrade
+/// @param destinationAmount The amount of destination tokens that will be downgraded 
+function downgrade(address _to, uint256 destinationAmount) external
+```
+
+### Events
+
+#### Upgrade
+
+Must trigger when tokens are upgraded using calls to ```upgrade```.
```

Comment:
```suggestion
Must trigger when tokens are upgraded.
```

Some implementations might try to weasel out of emitting this by using a different function.